src/main/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImpl.java

```diff
@@ -0,0 +1,283 @@
+        String failureMessage = KFSConstants.EMPTY_STRING;
+        int listIndex = 1;
+
+        if (StringUtils.isBlank(staleCheckRow.getBankCode())) {
+            failureMessage = appendFailureMessage(failureMessage, listIndex++, " Bank is blank. ");
+        }
+
+        if (StringUtils.isBlank(staleCheckRow.getCheckNumber())) {
+            failureMessage = appendFailureMessage(failureMessage, listIndex++, " Check Number is blank. ");
+        }
+
+        if (StringUtils.isBlank(staleCheckRow.getCheckTotalAmount())) {
+            failureMessage = appendFailureMessage(failureMessage, listIndex++, " Check Total Amount is blank. ");
+        }
+
+        if (ObjectUtils.isNull(checkReconciliation) || ObjectUtils.isNull(checkReconciliation.getCheckNumber())) {
+            failureMessage = appendFailureMessage(failureMessage, listIndex++, " Check Reconciliation does not exist in KFS. ");
+        } else {
+            if (!StringUtils.equalsIgnoreCase(staleCheckRow.getBankCode(), checkReconciliation.getBankCode())) {
+                String message = " The Bank Code in the file (" + staleCheckRow.getBankCode() + ") row does not match the bank on the Check (" +
+                        checkReconciliation.getBankCode() + "). ";
+                failureMessage = appendFailureMessage(failureMessage, listIndex++, message);
+            }
+
+            try {
+                Double checkTotalAmountParsed = NumberFormat.getNumberInstance(java.util.Locale.US).parse(staleCheckRow.getCheckTotalAmount()).doubleValue();
+                KualiDecimal checkTotalAmount = new KualiDecimal(checkTotalAmountParsed);
+                if (!checkTotalAmount.equals(checkReconciliation.getAmount())) {
+                    String message = " The Total Check Amount in the file row (" + checkTotalAmount.toString() + ") does not match the amount on the Check (" +
+                            checkReconciliation.getAmount().toString() + "). ";
+                    failureMessage = appendFailureMessage(failureMessage, listIndex++, message);
+                }
+            } catch (java.text.ParseException ex) {
+                String message = " Error converting total amount to decimal [" + staleCheckRow.getCheckTotalAmount() + "] " + ex.toString();
+                failureMessage = appendFailureMessage(failureMessage, listIndex++, message);
+            }
+
+            if (StringUtils.equalsIgnoreCase(checkReconciliation.getStatus(), CRConstants.STALE)) {
+                failureMessage = appendFailureMessage(failureMessage, listIndex++, " Check Status is already STALE.");
+            }
+        }
+
+        if (failureMessage.length() > 0) {
+            LOG.warn("validateStaleCheckBatchRow:" + failureMessage);
+            return failureMessage;
+        }
+        return StringUtils.EMPTY;
+    }
+
+    protected String appendFailureMessage(String failureMessage, int listIndex, String errorMessage) {
+        if (failureMessage.length() == 0) {

```

Comment:
You could use StringUtils.length to avoid NPEs, and inside this if statement, you don't need a concatenation